### ISSUE TITLE
Scaling issue with non-conventional monitor sizes.

### ISSUE THREAD

**Issue by kodiakbehr** [godotengine/godot#41969]
**Godot version:**
Godot 3.2.2.stable.official

**OS/device including version:**
Windows/PC  GLES3

**Issue description:**
A game with a native resolution of 2560x1440p, the stretch mode set to 2D, and the aspect mode to expand is displaying unusual outputs on a 3440 x 1440p monitor, below.  We suspect the issue lies with the engine, but would appreciate confirmation on this point.
![image (9)](https://user-images.githubusercontent.com/39195489/92835294-107ef600-f3a9-11ea-99d6-b8381ffec11c.jpg)

**Steps to reproduce:**
Run exported Godot project on a 3440 x 1440p monitor.

**Minimal reproduction project:**

**Comment by akien-mga** [comment]
A minimal reproduction project would be useful, as the issue is likely dependent on your project.godot configuration.

**Comment by Calinou** [comment]
Do you have fullscreen mode enabled? Is the window marked as resizable? I don't see anything in Godot that prevents it from supporting ultrawide monitors.

**Comment by kodiakbehr** [comment]
I've uploaded a monitor test reproduction project.  We tested this project on another user's ultrawide monitor for comparison, and got a different effect, although I don't think it is what I would expect.  Fullscreen mode is enabled and the window is marked as resizable.  Is there something else in the project configuration I should be concerned with?

![image](https://user-images.githubusercontent.com/39195489/93241987-20edf280-f754-11ea-940f-aeeab0795906.png)

[MonitorTestCase.zip](https://github.com/godotengine/godot/files/5226562/MonitorTestCase.zip)

**Comment by Calinou** [comment]
@kodiakbehr It looks expected to me on Godot 3.2.2. To get multiple aspect ratios to work correctly, you need to configure UI anchors so the UI will be centered.

**Comment by Calinou** [comment]
@kodiakbehr Have you been able to get multiple resolutions/aspect ratios working as per my comment?

**Comment by kodiakbehr** [comment]
Yes, things appear to be behaving as intended a single ultrawide monitor, however we recently got a report that the same file on another users ultrawide, which _also_ has an additional connected vertical monitor is displaying output that isn't what I would have expected.  It appears the issue is not the ultrawide, but the addition of a second monitor with an ultrawide?

Screenshot with the test program attached.

![image (3)](https://user-images.githubusercontent.com/39195489/94872176-e9509d00-0419-11eb-8550-26373c60be90.png)

**Comment by Calinou** [comment]
Is fullscreen enabled in that configuration? Does it work better if you also enable `borderless` in the Project Settings?

**Comment by kodiakbehr** [comment]
Good question.  Fullscreen is enabled, but borderless is off.  Will produce a borderless build and send that on.

**Comment by Eliepse** [comment]
Hi there,
This issue looks like one I had where the window does not fill my monitor when in fullscreen mode.
I have two monitors with the same resolution, but one with a scale factor of 125%.
I just fixed the issue by reseting the scale of my monitor to 100% (under Windows display settings).
I hope that help !

Maybe this is something that can be addressed to the Godot team ? Not sure if it has already been posted somewhere. Open to advices. :)

**Comment by Calinou** [comment]
> Maybe this is something that can be addressed to the Godot team ? Not sure if it has already been posted somewhere. Open to advices. :)

Make sure **Allow Hidpi** is enabled in the Project Settings.

**Comment by Eliepse** [comment]
> Enable **Allow Hidpi** in the Project Settings and make sure your project is designed to support [multiple resolutions](https://docs.godotengine.org/en/latest/tutorials/rendering/multiple_resolutions.html).

Thank you for the link.
Still, it does not fix the issue I had.
